ui/packages/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.view.js

```diff
@@ -13,26 +13,99 @@
-  childView: SearchFormView,
+const Item = styled.div`
+  display: inline-block;
+  padding: ${props => props.theme.mediumSpacing};
+  margin: ${props => props.theme.mediumSpacing};
+  width: calc(8 * ${props => props.theme.minimumButtonSize});
+  height: calc(4 * ${props => props.theme.minimumButtonSize});
+  text-align: left;
+  vertical-align: top;
+  position: relative;
+`
+
+const NewFormCircle = styled.div`
+  font-size: calc(3 * ${props => props.theme.largeFontSize});
+  padding-top: ${props => props.theme.minimumSpacing};
+`
+
+const NewSearchForm = ({ onClick }) => {
+  return (
+    <Item
+      className="is-button"
+      style={{ textAlign: 'center' }}
+      onClick={onClick}
+    >
+      <NewFormCircle className="fa fa-plus-circle" />
+      <h3 style={{ lineHeight: '2em' }}>New Search Form</h3>
+    </Item>
+  )
+}
+
+module.exports = Marionette.ItemView.extend({
   tagName: CustomElements.register('my-search-forms'),
   className: 'is-list is-inline has-list-highlighting',
-  initialize: function(options) {},
-  childViewOptions: function() {
-    return {
-      queryModel: this.options.queryModel,
-      collectionWrapperModel: this.options.collectionWrapperModel,
-      hideInteractionMenu: this.options.hideInteractionMenu,
-    }
+  initialize: function(options) {
+    this.model = this.options.collection
+    this.filter = this.options.filter
+    this.listenTo(this.model, 'add', this.render)
   },
-  filter: function(child) {
-    if (this.options.hideNewForm) {
-      return child.get('type') !== 'new-form'
+  template() {
+    return (
+      <React.Fragment>
+        {this.options.showNewForm ? (
+          <NewSearchForm onClick={this.handleNewForm.bind(this)} />
+        ) : null}
+        {this.model.filter(child => this.doFilter(child)).map(child => {
+          return (
+            <Item className="is-button" key={child.get('id')}>
+              <MarionetteRegionContainer
+                view={SearchFormView}
+                viewOptions={{
+                  model: child,
+                  queryModel: this.options.queryModel,
+                  collectionWrapperModel: this.options.collectionWrapperModel,
+                }}
+              />
+            </Item>
+          )
+        })}
+      </React.Fragment>
+    )
+  },
+  handleNewForm() {
+    this.options.queryModel.set({
+      type: 'new-form',
+      associatedFormModel: this.model,
+    })
+    user.getQuerySettings().set('type', 'new-form')
+    this.routeToSearchFormEditor('create')
+  },
+  doFilter(child) {
+    if (typeof this.options.filter !== 'function') {
+      // Show all children if no filter is provided
+      return true
     }
-    return child.get('type') !== 'basic' && child.get('type') !== 'text'
+    return this.options.filter(child)
+  },
+  routeToSearchFormEditor: function(newSearchFormId) {
+    const fragment = `forms/${newSearchFormId}`
+    wreqr.vent.trigger('router:navigate', {
```

Comment:
Are we still using `wreqr` to trigger navigation? Thought we were trying to stay clear of using it.

Reply:
Navigating between routes is one of the valid use cases for wreqr.